gradle-conjure/src/main/java/com/palantir/gradle/conjure/ConjureLocalPlugin.java

```diff
@@ -88,37 +89,37 @@ private void setupConjureJava(
                 subproj, "gitignoreConjureJava", subproj.getProjectDir(), ConjurePlugin.JAVA_GITIGNORE_CONTENTS);
 
-        project.getTasks().create("generateJava", ConjureLocalGenerateGenericTask.class, task -> {
-            task.setDescription("Generates Java bindings for remote Conjure definitions.");
-            task.setGroup(ConjurePlugin.TASK_GROUP);
-            // TODO(forozco): Automatically pass which category of code to generate
-            task.setOptions(() -> {
-                GeneratorOptions generatorOptions = optionsSupplier.get();
-                Preconditions.checkArgument(
-                        UNSAFE_JAVA_OPTIONS.stream().noneMatch(generatorOptions::has),
-                        "Unable to generate Java bindings since unsafe options were provided",
-                        generatorOptions.getProperties());
-
-                return generatorOptions;
-            });
-            task.setSource(conjureIrConfiguration);
-            task.setExecutablePath(extractJavaTask::getExecutable);
-            task.setOutputDirectory(subproj.file(ConjurePlugin.JAVA_GENERATED_SOURCE_DIRNAME));
-
-            generateConjure.dependsOn(task);
-            subproj.getTasks().getByName("compileJava").dependsOn(task);
-
-            ConjurePlugin.applyDependencyForIdeTasks(subproj, task);
-            task.dependsOn(gitignoreConjureJava);
-            task.dependsOn(extractJavaTask);
-
-            Task cleanTask = project.getTasks().findByName(ConjurePlugin.TASK_CLEAN);
-            cleanTask.dependsOn(project.getTasks().findByName("cleanGenerateJava"));
-            subproj.getDependencies().add("api", subproj);
-        });
+        TaskProvider<ConjureLocalGenerateGenericTask> generateJava = project.getTasks()
+                .register("generateJava", ConjureLocalGenerateGenericTask.class, task -> {
+                    task.setDescription("Generates Java bindings for remote Conjure definitions.");
+                    task.setGroup(ConjurePlugin.TASK_GROUP);
+                    // TODO(forozco): Automatically pass which category of code to generate
+                    task.setOptions(() -> {
+                        GeneratorOptions generatorOptions = optionsSupplier.get();
+                        Preconditions.checkArgument(
+                                UNSAFE_JAVA_OPTIONS.stream().noneMatch(generatorOptions::has),
+                                "Unable to generate Java bindings since unsafe options were provided",
+                                generatorOptions.getProperties());
+
+                        return generatorOptions;
+                    });
+                    task.setSource(conjureIrConfiguration);
+                    task.setExecutablePath(extractJavaTask::getExecutable);
+                    task.setOutputDirectory(subproj.file(ConjurePlugin.JAVA_GENERATED_SOURCE_DIRNAME));
+
+                    generateConjure.dependsOn(task);
```

Comment:
it isn't safe to add dependencies on a task within its configuration block if you are registering it lazily since the config lambda is only ever executed if the task is already on the dependency graph. You'll need to pull this out

Reply:
nice catch!